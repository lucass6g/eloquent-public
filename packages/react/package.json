{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
