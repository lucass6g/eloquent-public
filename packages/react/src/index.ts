export {Switch, Case, Default} from './components/conditional-switch'