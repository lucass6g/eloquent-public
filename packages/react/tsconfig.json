{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
