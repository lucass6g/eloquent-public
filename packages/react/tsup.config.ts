import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})