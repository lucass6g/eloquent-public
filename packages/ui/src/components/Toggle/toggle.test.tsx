import { render, screen } from '@testing-library/react';
import { it, describe, expect } from "vitest"
import { Toggle } from '.';
import { ComponentIcon } from 'lucide-react';

describe('Toggle Component', () => {

    it('renders ToggleWithText story', () => {
        const component = render(<Toggle variant='default'><ComponentIcon data-testid="svg-icon" /> Label </Toggle>);
        expect(screen.getByText('Label')).toBeInTheDocument();
        expect(screen.getByTestId('svg-icon')).toBeInTheDocument();
        expect(screen.getByRole('button')).toBeInTheDocument();
        expect(component.baseElement).toMatchSnapshot();
    });

    it('renders ToggleDefault story', () => {
        render(<Toggle variant='outline' size='sm'><ComponentIcon /> </Toggle>);
        expect(screen.getByRole('button')).toBeInTheDocument();
    });
});