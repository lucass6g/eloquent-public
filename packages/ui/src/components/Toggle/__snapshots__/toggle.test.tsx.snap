// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toggle Component > renders ToggleWithText story 1`] = `
<body>
  <div>
    <button
      aria-pressed="false"
      class="inline-flex items-center justify-between rounded-[6px] text-sm font-medium transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground [&>svg]:mr-2 bg-transparent h-10 px-3 w-[90px]"
      data-state="off"
      type="button"
    >
      <svg
        class="lucide lucide-component"
        data-testid="svg-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 8.5 9 12l-3.5 3.5L2 12l3.5-3.5Z"
        />
        <path
          d="m12 2 3.5 3.5L12 9 8.5 5.5 12 2Z"
        />
        <path
          d="M18.5 8.5 22 12l-3.5 3.5L15 12l3.5-3.5Z"
        />
        <path
          d="m12 15 3.5 3.5L12 22l-3.5-3.5L12 15Z"
        />
      </svg>
       Label 
    </button>
  </div>
</body>
`;
